clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/AbstractPartitionStateModel.java

```diff
@@ -46,6 +46,15 @@ public abstract class AbstractPartitionStateModel extends StateModel {
   private static final int RETRY_COUNT = 5;
   private static final int RETRY_DURATION_MS = 1000;
   private static final int WAIT_PARTITION_ACCESSOR_TIME_OUT_MS = (int) TimeUnit.MINUTES.toMillis(5);
+  // The drop partition consumer action may take longer than expected and eventually timeout if:
+  // 1. The corresponding ingestion task is stuck and unable to process consumer actions.
+  // 2. The corresponding ingestion task died due to errors.
+  // We want to wait for drop partition because consumer actions are executed asynchronously w.r.t. Helix state
+  // transitions. Currently, we open a partition synchronously during OFFLINE->STANDBY. This can be problematic and have
+  // race conditions if we don't wait on the drop partition during X->DROPPED. This is because Helix can send the
+  // following state transitions right after each other for the same partition to the same node:
+  // X->DROPPED, OFFLINE->STANDBY
+  protected static final long WAIT_DROP_PARTITION_TIME_OUT_MS = TimeUnit.MINUTES.toMillis(60);
```

Comment:
Is 60 minutes sufficient?

Reply:
Normally it shouldn't take that long right? IMO if the SIT is stuck or slow we will have other alerts/signal firing before 60 minutes.

Reply:
Makes sense to me.